src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbComposite.java

```diff
@@ -0,0 +1,169 @@
+import com.amazonaws.services.dynamodbv2.model.DeleteTableResult;
+import com.amazonaws.services.dynamodbv2.model.DescribeTableRequest;
+import com.amazonaws.services.dynamodbv2.model.DescribeTableResult;
+import com.amazonaws.services.dynamodbv2.model.GetItemRequest;
+import com.amazonaws.services.dynamodbv2.model.GetItemResult;
+import com.amazonaws.services.dynamodbv2.model.ListTablesResult;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import com.amazonaws.services.dynamodbv2.model.PutItemResult;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.amazonaws.services.dynamodbv2.model.QueryResult;
+import com.amazonaws.services.dynamodbv2.model.ScanRequest;
+import com.amazonaws.services.dynamodbv2.model.ScanResult;
+import com.amazonaws.services.dynamodbv2.model.UpdateItemRequest;
+import com.amazonaws.services.dynamodbv2.model.UpdateItemResult;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * An {@link MtAmazonDynamoDb} that determines which of multiple {@link MtAmazonDynamoDb}s to delegate to based on
+ * context.
+ *
+ * <p>TODO: support backup-related functionality
+ */
+public class MtAmazonDynamoDbComposite extends MtAmazonDynamoDbBase {
+
+    private final Collection<MtAmazonDynamoDbBase> delegates;
+    private final Supplier<MtAmazonDynamoDbBase> fromContextToDelegate;
+    private final Function<String, MtAmazonDynamoDbBase> fromPhysicalTableNameToDelegate;
+
+    public MtAmazonDynamoDbComposite(Collection<MtAmazonDynamoDbBase> delegates,
+                                     Supplier<MtAmazonDynamoDbBase> fromContextToDelegate,
+                                     Function<String, MtAmazonDynamoDbBase> fromPhysicalTableNameToDelegate) {
+        super(validateDelegatesAndGetFirst(delegates).getMtContext(),
+            delegates.iterator().next().getAmazonDynamoDb(),
+            delegates.iterator().next().getMeterRegistry(),
+            delegates.iterator().next().getScanTenantKey(),
+            delegates.iterator().next().getScanVirtualTableKey());
+        this.delegates = delegates;
+        this.fromContextToDelegate = fromContextToDelegate;
+        this.fromPhysicalTableNameToDelegate = fromPhysicalTableNameToDelegate;
+    }
+
+    private static MtAmazonDynamoDbBase validateDelegatesAndGetFirst(Collection<MtAmazonDynamoDbBase> delegates) {
+        checkArgument(delegates != null && !delegates.isEmpty(), "Must provide at least one delegate");
+        Iterator<MtAmazonDynamoDbBase> iterator = delegates.iterator();
+        MtAmazonDynamoDbBase first = iterator.next();
+        while (iterator.hasNext()) {
+            MtAmazonDynamoDbBase next = iterator.next();
+            checkArgument(Objects.equals(first.getMtContext(), next.getMtContext()),
+                "Delegates must share the same mt context provider");
```

Comment:
So if we used different instances of the same context provider, we'd have to make sure to override `equals`?

Reply:
Yeah we'd have to do that. Right now with the DynamoDbConfig changes I propose in the other PR, both the random partitioning client and the hash partitioning client have the same instance of MtAmazonDynamoDbContextProvider.